pkg/health/membergarbagecollector/membergarbagecollector.go

```diff
@@ -0,0 +1,174 @@
+	"github.com/gardener/etcd-backup-restore/pkg/wrappers"
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	coordv1 "k8s.io/api/coordination/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	etcdclient "github.com/gardener/etcd-backup-restore/pkg/etcdutil/client"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	podNamespace = "POD_NAMESPACE"
+	podName      = "POD_NAME"
+)
+
+// MGCChecker garbage collects the members from the etcd cluster if its corresponding pod is not present. Used by Gomock to generate mock functions
+type MGCChecker interface {
+	RemoveSuperfluousMembers(ctx context.Context) error
+}
+
+// MemberGarbageCollector contains information to regularly cleanup superfluous ETCD members
+type MemberGarbageCollector struct {
+	logger                *logrus.Entry
+	gcTimer               *time.Timer
+	k8sClient             client.Client
+	podNamespace          string
+	stsName               string
+	etcdConnectionTimeout wrappers.Duration
+}
+
+// NewMemberGarbageCollector returns the member garbage collect object
+func NewMemberGarbageCollector(logger *logrus.Entry, clientSet client.Client, etcdTimeout wrappers.Duration) (*MemberGarbageCollector, error) {
+	if clientSet == nil {
+		return nil, &errors.EtcdError{
+			Message: "nil kubernetes clientset passed, can not create etcd member garbage collector",
+		}
+	}
+	namespace, err := utils.GetEnvVarOrError(podNamespace)
+	if err != nil {
+		logger.Fatalf("POD_NAMESPACE env var not present: %v", err)
+	}
+	podName, err := utils.GetEnvVarOrError(podName)
+	if err != nil {
+		logger.Fatalf("POD_NAME env var not present: %v", err)
+	}
+	return &MemberGarbageCollector{
+		logger:                logger.WithField("actor", "etcd-membergc"),
+		k8sClient:             clientSet,
+		podNamespace:          namespace,
+		stsName:               podName[:strings.LastIndex(podName, "-")],
+		etcdConnectionTimeout: etcdTimeout,
+	}, nil
+}
+
+// RunMemberGarbageCollectorPeriodically periodically calls the member garbage collector
+func RunMemberGarbageCollectorPeriodically(ctx context.Context, hconfig *brtypes.HealthConfig, logger *logrus.Entry, etcdConfig *brtypes.EtcdConnectionConfig) {
+	clientSet, err := miscellaneous.GetKubernetesClientSetOrError()
+	if err != nil {
+		logger.Fatalf("failed to create kubernetes clientset: %v", err)
+		return
+	}
+	mgc, err := NewMemberGarbageCollector(logger, clientSet, etcdConfig.ConnectionTimeout)
+	if err != nil {
+		logger.Errorf("failed to initialize new etcd member garbage collector: %v", err)
+		return
+	}
+	mgc.gcTimer = time.NewTimer(hconfig.MemberGCDuration.Duration)
+	defer mgc.gcTimer.Stop()
+	mgc.logger.Info("Started etcd member garbage collector")
+
+	//Create etcd client to get etcd cluster member list
+	clientFactory := etcdutil.NewFactory(*etcdConfig)
+	etcdCluster, err := clientFactory.NewCluster()
+	if err != nil {
+		mgc.logger.Errorf("Failed to create etcd cluster client: %v", err)
```

Comment:
```suggestion
		mgc.logger.Errorf("failed to create etcd cluster client: %v", err)
```
If we failed to create `etcdCluster` then I think it is better to return. WDYT?